src/launcher/nix-launch.ts

```diff
@@ -94,9 +99,67 @@ export default async function launch(config: ConfigWithRuntime, osConfig: OsConf
             args.push('--v=1');
             args.push('--attach-console');
         }
+        if (typeof config.services !== undefined && config.services != null) {
+            // tslint:disable-next-line:no-use-before-declare
+            await startServices(config.services);
+        }
         return spawn(runtimePath, args);
     } catch (e) {
         console.error('Failed to launch\n', e);
         throw e;
     }
 }
+
+interface AppDirectoryEntry {
+    manifest_url: string;
+}
+
+const fetchServiceDefinition = async (url: string): Promise<AppDirectoryEntry> => {
+    const lookupUrl = parse(url);
+    const protocol = await import(lookupUrl.protocol.slice(0, -1));
+    const res = await new Promise<string>(async (resolve, reject) => {
```

Comment:
Use `get` from ./utils instead

Reply:
that function naively presumes HTTPS which isn't a bad thing in a general but could make dev harder where certs are rarely used - unless i missing something.